sdk/java/dagger-java-sdk/src/main/java/io/dagger/client/exception/DaggerExceptionUtils.java

```diff
@@ -0,0 +1,96 @@
+package io.dagger.client.exception;
+
+import static io.dagger.client.exception.DaggerExceptionConstants.CMD_KEY;
+import static io.dagger.client.exception.DaggerExceptionConstants.ENHANCED_MESSAGE;
+import static io.dagger.client.exception.DaggerExceptionConstants.EXIT_CODE_KEY;
+import static io.dagger.client.exception.DaggerExceptionConstants.FULL_MESSAGE;
+import static io.dagger.client.exception.DaggerExceptionConstants.SIMPLE_MESSAGE;
+import static io.dagger.client.exception.DaggerExceptionConstants.STDERR_KEY;
+import static io.dagger.client.exception.DaggerExceptionConstants.STDOUT_KEY;
+import static io.dagger.client.exception.DaggerExceptionConstants.TYPE_KEY;
+
+import io.smallrye.graphql.client.GraphQLError;
+import jakarta.json.JsonArray;
+import jakarta.json.JsonValue;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.StringUtils;
+
+public class DaggerExceptionUtils {
+
+  private DaggerExceptionUtils() {}
+
+  public static Object getExtensionValueByKey(GraphQLError error, String key) {
+    if (error == null || StringUtils.isBlank(key) || error.getExtensions() == null) {
+      return null;
+    }
+
+    return error.getExtensions().getOrDefault(key, null);
+  }
+
+  public static String getPath(GraphQLError error) {
+    return StringUtils.join(error.getPath(), ".");
+  }
+
+  public static String getCmd(GraphQLError error) {
+    Object cmdList = getExtensionValueByKey(error, CMD_KEY);
+    if (cmdList != null && cmdList instanceof JsonArray array) {
+      return array.stream()
+          .map(JsonValue::toString)
+          .collect(Collectors.joining(" "))
+          .replace("\"", "");
```

Comment:
I think we should just return a `List<String>` and not try to join them. That way we are similar to what Go SDK is doing.

Reply:
👍